File: containsDuplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        dict = {}
        for x in nums:
            try:
                dict[x] += 1
                if dict[x] == 2:
                    return True
            except: 
                dict[x] = 1
        return False
            

#https://leetcode.com/problems/contains-duplicate/
#Given an integer array nums, return true if any value appears at least twice in 
# the array, and return false if every element is distinct.

File: containsDuplicate.cpp
using namespace std;
#include <vector>
#include <set>

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> seen;
        int n = nums.size();

        for(int elt: nums){
            if(seen.find(elt) != seen.end()){
                return true;
            }
            seen.insert(elt);
        }
        return false;
    }
};

int main(){
    
}
/*
https://leetcode.com/problems/contains-duplicate/description/

Given an integer array nums, return true if any value appears at least twice in the array,
 and return false if every element is distinct.
*/